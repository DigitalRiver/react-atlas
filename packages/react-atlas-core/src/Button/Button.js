import React from "react";
import PropTypes from "prop-types";
import {IconCore} from './../Icon';
import cx from "classnames";

/**
 * A Generic button component.
 */
class Button extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      onClick,
      className,
      children,
      disabled,
      large,
      small,
      primary,
      secondary,
      warning,
      error,
      link,
      outline,
      icon,
      type,
      href,
      style
    } = this.props;

    let mainStyle = "button";
    if (primary) {
      mainStyle = "primary";
    } else if (secondary) {
      mainStyle = "secondary";
    } else if (warning) {
      mainStyle = "warning";
    } else if (error) {
      mainStyle = "error";
    } else if (link) {
      mainStyle = "link";
    } else {
      mainStyle = "default_btn";
    }

    if (outline) {
      mainStyle += "_outline";
    }

    /**
     * Use cx to construct a CSS class object.
     * cx will automatically remove falsy values
     * for us. This way only passed props will be
     * transformed into CSS classes.
     */
    const classes = cx(
      {
        "disabled": disabled,
        "large": large,
        "small": small,
        "button": true
      },
      mainStyle
    );

    let text = children;
    let iconClass = false;

    /* Just use the icon instead of text when an icon
     * is passed and no children(text) is set. */
    if (icon && children === "Default Button") {
      text = null;
    } else if (icon) {
      text = children;
      iconClass = "ra_Button__icon-left";
    }

    let btn = <button
          onClick={onClick}
          className={cx(className)}
          styleName={classes}
          style={style}
          type={type}
          href={href}
        >
          {icon ? <IconCore icon={icon} /> : null}
          {text}
        </button>
    ;

    let renderButton = btn;
    if (href) {
      renderButton = <a href={href}>{btn}</a>;
    }

    return renderButton;
  }
}

Button.propTypes = {
  /** HTML5 button type, eg submit, button, etc */
  "type": PropTypes.string,
  /** The URL to link to. */
  "href": PropTypes.string,
  /**
   * Define a mini button.
   *
   */
  "children": PropTypes.node,
  /**
   * Click event handler.
   */
  "onClick": PropTypes.func,
  /** An Object, array, or string of CSS classes to apply to button.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   *Determines if the button is disabled.
   * @examples <Button disabled/>
   */
  "disabled": PropTypes.bool,
  /**
   * use outline styled button
   * @examples <Button outline>
   */
  "outline": PropTypes.bool,
  /**
   * use primary style button
   * @examples <Button primary>
   */
  "primary": PropTypes.bool,
  /**
   * use secondary style button
   * @examples <Button secondary>
   */
  "secondary": PropTypes.bool,
  /**
   * use warning style button
   * @examples <Button warning>
   */
  "warning": PropTypes.bool,
  /**
   * use error style button
   * @examples <Button error>
   */
  "error": PropTypes.bool,
  /**
   * use link style button
   * @examples <Button link>
   */
  "link": PropTypes.bool,
  /**
   * use large style button
   * @examples <Button large>
   */
  "large": PropTypes.bool,
  /**
   * use small style button
   * @examples <Button small>
   */
  "small": PropTypes.bool,

  /**
   * The class name of the icon you want to set.
   */
  "icon": PropTypes.string,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object
};

Button.defaultProps = {
  "children": "Default Button",
  "outline": false,
  "primary": false,
  "secondary": false,
  "warning": false,
  "error": false,
  "link": false,
  "large": false,
  "small": false,
  "disabled": false,
  "type": "button"
};

export default Button;
