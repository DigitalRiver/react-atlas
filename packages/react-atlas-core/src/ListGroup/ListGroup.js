import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class ListGroup extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const { children, className, divider, id, style, title } = this.props;

    const groupStyles = cx({
      divider
    });

    return (
      <div id={id} style={style} className={className} styleName={groupStyles}>
        <div styleName="groupTitle">{title}</div>
        {children}
      </div>
    );
  }
}

ListGroup.propTypes = {
  /**
   * Text, any HTML element, or React Component.
   */
  "children": PropTypes.node,
  /** An Object, array, or string of CSS classes to apply to Input. */
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /** Adds a horizontal rule between the ListGroup and subsequent children. */
  "divider": PropTypes.bool,
  /**
   * Defines an id for the input.
   * @examples '<Text as="text" name="test"/>'
   */
  "id": PropTypes.string,
  /** Pass inline styling here. */
  "style": PropTypes.object,
  /** Adds a title to the ListGroup. */
  "title": PropTypes.string
};

ListGroup.defaultProps = {
  "divider": true
};

export default ListGroup;
