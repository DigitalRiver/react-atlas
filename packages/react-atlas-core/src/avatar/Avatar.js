import React, { PropTypes } from "react";
import themeable from 'react-themeable';

/**
 * Avatar component creates a circular area where an image, letter or icon/glyphicon can be presented. Great for user profiles and lists.
 *
 * **NOTE**: children will always take precedence over props passed into component.
 */
 class Avatar extends React.Component {
   constructor(props) {
     super(props);

     var image;

     if(this.props.image == undefined) {
       if(this.props.defaultImage == undefined) {
         image = null;
       } else {
         image = this.props.defaultImage;
       }
     } else {
       image = this.props.image;
     }

     this.state = {image: image, title: this.props.title, icon: this.props.icon};
   }

   handleBadImage() {
     if(this.props.defaultImage === this.state.image) {
        this.setState({image: null});
        return;
     }

     this.setState({image: this.props.defaultImage});
   }

   render() {
     const theme = themeable(this.props.theme);
     var children = this.props.children;
     let kids = children;
     if (React.Children.count(children) === 1 && typeof children === "string"){
       kids = <span {...theme(4, 'letter')}>{children[0]}</span>;
     }

     let avatar = null;
     var icon = this.state.icon;
     var image = this.state.image;
     var title = this.state.title;

     if (typeof image === "string") {
       avatar = <img src={image} title={title} onError={this.handleBadImage.bind(this)} {...theme(3, 'image')} />;
     } else if (image) {
       avatar = image;
     } else if (icon) {
       avatar = icon;
     } else if (title) {
       avatar = <span {...theme(2, 'letter')}>{title[0]}</span>;
     }
     return (
       <div {...theme(1, 'avatar')}>
         {kids}
         {avatar}
       </div>
     );
   }
 }

Avatar.propTypes = {
  /**
   * Children should be either a string, an icon/glyphicon, or an image tag.
   * @examples "SomeName", <SomeIcon />, <img src="/path/to/image.jpg"/>
   */
  children: PropTypes.node,
  /**
   * A css class name that will be appended to the wrapping div around the avatar.
   */
  className: PropTypes.string,
  /**
   * For displaying an icon/glphyicon. Normally these will be another component or an element with a class on it.
   * @examples <GithubIcon />, <i class="fa fa-github"></i>
   */
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  /**
   * Path to an image
   * @examples "http://path.to/an/image.jpg"
   */
  image: PropTypes.string,
  /**
   * A string. Avatar will use First letter of the string.
   * @examples "Nathan" will output "N"
   */
  title: PropTypes.string,

  /**
   * A URL to a image that is displayed when the main image fails to load.
   */
  defaultImage: PropTypes.string
};

Avatar.defaultProps = {
  className: '',
  icon: ''
};

Avatar.styleguide = {
  category: 'Avatar',
  index: '2.1',
  example: `
  <section>
    <h5>Avatars</h5>
    {/* background color change */}
    <Avatar style={{backgroundColor: 'deepskyblue'}} >
      <i className="fa fa-github"></i>
    </Avatar>
    {/* title prop gets truncated to 1st letter */}
    <Avatar title="Nathan" />
    {/* icon beats title */}
    <Avatar title="Nathan" icon={<i className="fa fa-github"></i>} />
    {/* image beats icon */}
    <Avatar
      icon={<i className="fa fa-github"></i>}
      image="cat.jpg"
    />
    {/* image beats title */}
    <Avatar title="Javier" image="cat.jpg" />
    {/* child beats parameters */}
    <Avatar title="Nathan" image="cat.jpg">
      <i className="fa fa-github"></i>
    </Avatar>
    <Avatar title="Nathan" icon={<i className="fa fa-github"></i>}>
      <img src="cat.jpg"/>
    </Avatar>
    {/* child string gets truncated to 1st letter */}
    <Avatar>Nathan</Avatar>
    {/* child should be <i>, <svg>, <img>, or string */}
    <Avatar><i className="fa fa-github"></i></Avatar>
  </section>
  `
};
export default Avatar;
