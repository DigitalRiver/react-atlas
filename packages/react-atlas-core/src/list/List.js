import React from "react";
import cx from 'classNames';

/**
 * Wrapper around `<ul>` element to apply styling to lists. Usually paired with `<ListItem>` and `<ListText>` components.
 */
const List = ({ children, ...props }) => {
  return (
    <ul {...props} styleName={cx("list")}>
      {children}
    </ul>
  );
};

List.propTypes = {
  "children": React.PropTypes.node,
  "className": React.PropTypes.string
};

List.styleguide = {
  "category": "Lists",
  "index": "7.1",
  "example": 
    `
<section>
  <h5>Simple List</h5>
  <div style={{
    display: "inline-block",
    minWidth: 340
  }}>
    <List>
      <li>Inbox</li>
      <li>Outbox</li>
      <li>Trash</li>
      <li>Spam</li>
    </List>
  </div>
  
  <h5>Complex List using ListItem and ListText</h5>
  <div style={{
      display: "inline-block",
      minWidth: 340
  }}>
    <h5>Contacts</h5>
    <List>
      <ListItem>
        <Avatar image="jjj.jpg" />
        <ListText>
          Some Guy
          <Hint> Works at Some Place</Hint>
        </ListText>
      </ListItem>

      <ListItem>
        <Avatar image="gates.jpg" />
        <ListText>
          Other Person
          <Hint> Works at Other Place</Hint>
        </ListText>
      </ListItem>

      <ListItem>
        <Avatar image="chillgirl.jpeg" />
        <ListText>
          Chill Girl
          <Hint> Works at Chill Place</Hint>
        </ListText>
      </ListItem>

    </List>
  </div>
</section>
`
  
};

export default List;
