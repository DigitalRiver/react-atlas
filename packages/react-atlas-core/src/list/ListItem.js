import React from "react";
import themeable from "react-themeable";

/**
 * Wrapper around `<li>` element that applies styles.
 */
const ListItem = ({ children, ...props }) => {
  const theme = themeable(props.theme);

  return (
    <li {...props} {...theme(1, "item")}>
      {children}
    </li>
  );
};

ListItem.propTypes = {
  "className": React.PropTypes.string,
  "children": React.PropTypes.node,
  "theme": React.PropTypes.object
};

ListItem.defaultProps = { "className": "" };

ListItem.styleguide = {
  "category": "Lists",
  "index": "7.2",
  "example": 
    `
<section>
  <h5>Simple List with ListIem</h5>
  <div style={{
    display: "inline-block",
    minWidth: 340
  }}>
    <ListItem caption="Inbox">Inbox</ListItem>
    <ListItem caption="Outbox">Outbox</ListItem>
    <ListItem caption="Trash">Trash</ListItem>
    <ListItem caption="Spam">Spam</ListItem>
  </div>
</section>
`
  
};

export default ListItem;
