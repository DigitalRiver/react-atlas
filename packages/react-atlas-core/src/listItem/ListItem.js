import React from "react";
import cx from 'classNames';

/**
 * Wrapper around `<li>` element that applies styles.
 */
const ListItem = ({ children, ...props }) => {
  return (
    <li {...props} styleName={cx("item")}>
      {children}
    </li>
  );
};

ListItem.propTypes = {
  "className": React.PropTypes.string,
  "children": React.PropTypes.node
};

ListItem.styleguide = {
  "category": "Lists",
  "index": "7.2",
  "example": 
    `
<section>
  <h5>Simple List with ListIem</h5>
  <div style={{
    display: "inline-block",
    minWidth: 340
  }}>
    <ListItem caption="Inbox">Inbox</ListItem>
    <ListItem caption="Outbox">Outbox</ListItem>
    <ListItem caption="Trash">Trash</ListItem>
    <ListItem caption="Spam">Spam</ListItem>
  </div>
</section>
`
  
};

export default ListItem;
