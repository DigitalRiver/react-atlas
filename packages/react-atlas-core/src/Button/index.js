import { default as ButtonCore } from "./Button";
export { ButtonCore };
