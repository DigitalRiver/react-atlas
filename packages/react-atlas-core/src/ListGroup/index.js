import { default as ListGroupCore } from "./ListGroup";
export { ListGroupCore };