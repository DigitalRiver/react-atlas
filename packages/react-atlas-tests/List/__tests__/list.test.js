import React from "react";
import { mount } from "enzyme";
import { ListCore } from "../../../react-atlas-core/src/List/index";
import { ListGroupCore } from "../../../react-atlas-core/src/ListGroup/index";
import { ListItemCore } from "../../../react-atlas-core/src/ListItem/index";
import { AvatarCore } from "../../../react-atlas-core/src/Avatar/index";
import { SwitchCore } from "../../../react-atlas-core/src/Switch/index";
import { ButtonCore } from "../../../react-atlas-core/src/Button/index";
import renderer from "react-test-renderer";

describe("Test List component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ListCore border>
          <ListGroupCore title="Dinosaurs">
            <ListItemCore
              leftItem={<AvatarCore title="Blue" />}
              rightItem={<SwitchCore />}
            >
              <ButtonCore link>Blue</ButtonCore>
            </ListItemCore>
            <ListItemCore
              leftItem={<AvatarCore title="Earl" />}
              rightItem={<SwitchCore />}
            >
              <ButtonCore link>Earl Sinclair</ButtonCore>
            </ListItemCore>
          </ListGroupCore>
        </ListCore>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
