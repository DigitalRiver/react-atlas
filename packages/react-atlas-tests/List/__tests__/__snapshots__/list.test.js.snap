// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List component renders correctly 1`] = `
<div
  className={undefined}
  id={undefined}
  style={undefined}
  styleName="list border"
>
  <div
    className={undefined}
    id={undefined}
    style={undefined}
    styleName="divider"
  >
    <div
      styleName="groupTitle"
    >
      Dinosaurs
    </div>
    <div
      className={undefined}
      id={undefined}
      onClick={undefined}
      style={undefined}
      styleName="listItem"
    >
      <div
        styleName="inlineBlock"
      >
        <div
          className=""
          style={undefined}
          styleName="avatar"
          title="Blue"
        >
          <span
            styleName="letter"
          >
            B
          </span>
        </div>
      </div>
      <div
        styleName="inlineBlock"
      >
        <button
          className=""
          href={undefined}
          id={undefined}
          name={undefined}
          onClick={undefined}
          style={undefined}
          styleName="button link link_hover"
          tabIndex={0}
          type="button"
        >
          Blue
        </button>
      </div>
      <div
        styleName="rightItem"
      >
        <div
          className=""
          styleName="switch"
        >
          <div
            onClick={[Function]}
            style={undefined}
            styleName="sliderMedium sliderEnabledOffColor"
          >
            <input
              checked={false}
              className=""
              id={undefined}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              style={undefined}
              styleName="inputMedium"
              type="checkbox"
            />
            <div
              style={
                Object {
                  "background": undefined,
                }
              }
              styleName="handleMedium handleEnabledColor"
            />
            <div
              style={
                Object {
                  "background": undefined,
                }
              }
              styleName="onColorMedium"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className={undefined}
      id={undefined}
      onClick={undefined}
      style={undefined}
      styleName="listItem"
    >
      <div
        styleName="inlineBlock"
      >
        <div
          className=""
          style={undefined}
          styleName="avatar"
          title="Earl"
        >
          <span
            styleName="letter"
          >
            E
          </span>
        </div>
      </div>
      <div
        styleName="inlineBlock"
      >
        <button
          className=""
          href={undefined}
          id={undefined}
          name={undefined}
          onClick={undefined}
          style={undefined}
          styleName="button link link_hover"
          tabIndex={0}
          type="button"
        >
          Earl Sinclair
        </button>
      </div>
      <div
        styleName="rightItem"
      >
        <div
          className=""
          styleName="switch"
        >
          <div
            onClick={[Function]}
            style={undefined}
            styleName="sliderMedium sliderEnabledOffColor"
          >
            <input
              checked={false}
              className=""
              id={undefined}
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              style={undefined}
              styleName="inputMedium"
              type="checkbox"
            />
            <div
              style={
                Object {
                  "background": undefined,
                }
              }
              styleName="handleMedium handleEnabledColor"
            />
            <div
              style={
                Object {
                  "background": undefined,
                }
              }
              styleName="onColorMedium"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
